assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -297,71 +286,97 @@ export default function UserDimensionsPieChart( { dimensionName, dimensionValue,
-						chartID={ chartID }
-						chartType="pie"
-						options={ options }
-						data={ dataMap || [] }
-						loadHeight={ 50 }
-						onReady={ onReady }
-					/>
-
+			<div className={ classnames(
+				'googlesitekit-widget--analyticsAllTraffic__dimensions-chart',
+				{
+					'googlesitekit-widget--analyticsAllTraffic__slice-selected': !! dimensionValue,
+					'googlesitekit-widget--analyticsAllTraffic__selectable': selectable,
+				}
+			) }>
+				{ /* eslint-disable-next-line jsx-a11y/mouse-events-have-key-events */ }
+				<GoogleChartV2
+					chartType="PieChart"
+					data={ dataMap || [] }
+					getChartWrapper={ ( chartWrapper ) => {
+						chartWrapperRef.current = chartWrapper;
+						// Forces a re-render of the component to re-run useEffect hooks.
+						// If this is not called and the chart is updated while the mouse cursor
+						// is already over the chart, it won't properly run the `onmousenter` event
+						// and the "selectable" slices of the pie chart won't be enabled consistently.
+						setSelectable( null );
+					} }
+					height="368px"
+					loaded={ loaded }
+					loadingHeight="300px"
+					loadingWidth="300px"
+					onMouseOut={ onMouseOut }
+					onMouseOver={ onMouseOver }
+					onReady={ onReady }
+					onSelect={ onSelect }
+					options={ options }
+					width="100%"
+				>
 					<div
 						className="googlesitekit-widget--analyticsAllTraffic__dimensions-chart-title"
 						dangerouslySetInnerHTML={ title }
 					/>
-				</div>
-			</div>
+				</GoogleChartV2>
+
+				<div className="googlesitekit-widget--analyticsAllTraffic__legend">
+					{ loaded && dataMap?.slice( 1 ).map( ( [ label ], i ) => {
+						const isActive = label === dimensionValue;
+						const sliceColor = slices[ i ]?.color;
+						const isOthers = __( 'Others', 'google-site-kit' ) === label;
+
+						return (
+							<Link
+								key={ label }
+								onClick={ () => onLegendClick( i ) }
+								className={ classnames(
+									'googlesitekit-widget--analyticsAllTraffic__legend-slice',
+									{
+										'googlesitekit-widget--analyticsAllTraffic__legend-active': isActive,
+										'googlesitekit-widget--analyticsAllTraffic__legend-others': isOthers,
+									}
+								) }
+							>
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: sliceColor } } />
+
+								<span className="googlesitekit-widget--analyticsAllTraffic__label" data-label={ label }>
+									{ label }
+								</span>
+
+								<span className="googlesitekit-widget--analyticsAllTraffic__underlay" style={ { backgroundColor: sliceColor } } />
+							</Link>
+						);
+					} ) }
+
+					{ ! loaded && (
+						<Fragment>
+							<span className="googlesitekit-widget--analyticsAllTraffic__legend-slice">
+								<span className="googlesitekit-widget--analyticsAllTraffic__dot" style={ { backgroundColor: '#ccc' } } />
+								<PreviewBlock height="18px" width="68px" shape="square" />
+							</span>
```

Comment:
PreviewBlock renders a div which means that we will have divs nested in spans. Let's use `div` tags for legend preview items instead.

Reply:
Good catch, thanks!